process/block/preprocess/sovereignChainTransactions.go

```diff
@@ -151,9 +157,70 @@ func (sct *sovereignChainTransactions) shouldContinueProcessingScheduledTx(
 
 	addressHasEnoughBalance := sct.hasAddressEnoughInitialBalance(tx)
 	if !addressHasEnoughBalance {
+		log.Debug("address has not enough initial balance", "sender", sct.pubkeyConverter.Encode(tx.SndAddr))
 		mbInfo.schedulingInfo.numScheduledTxsWithInitialBalanceConsumed++
 		return nil, nil, false
 	}
 
 	return tx, miniBlock, true
 }
+
+func (sct *sovereignChainTransactions) shouldSkipMiniBlock(miniBlock *block.MiniBlock) bool {
```

Comment:
I think only the second check in IF is needed. Like sct.IsMiniblockCorrect - otherwise you should not skip. Sovereign is only one shard.

Reply:
Done